### PR TITLE
Add append operation example

### PR DESCRIPTION
**Pull request by JinnyYi**
Add append operation example in ref: https://github.com/aos-dev/go-storage/issues/529.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,3 +8,4 @@
 - [Read a file](read.go)
 - [Read a range of a file](read.go)
 - [Read a file with callback](read.go)
+- [Append an existing or new file](append.go)
```
Comment (Xuanwo) [inline README.md]: We need to split them into two examples:

- Append to a new file (create_append + write_append)
- Append to an existing file (stat + write_append)

File: append.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
+package example
+
+import (
+	"bytes"
+	"github.com/aos-dev/go-storage/v3/pkg/randbytes"
+	"github.com/aos-dev/go-storage/v3/types"
+	"io"
+	"io/ioutil"
+	"log"
+	"math/rand"
```
Comment (Xuanwo) [inline append.go:8]:
```go
import (
	"bytes"
	"io"
	"io/ioutil"
	"log"
	"math/rand"

	"github.com/aos-dev/go-storage/v3/pkg/randbytes"
	"github.com/aos-dev/go-storage/v3/types"
)
```

Usually, we organize our imports in the following way:

```
import (
	<golang stdlib>

	<3rd lib>

	<self lib>
)
```

File: append.go [hunk 3]
```diff
@@ -48,3 +48,46 @@ func Append(appender types.Appender, path string) {
...
+	}
+
+	// appendable object's mode must be Appendable and Readable
+	if !o.Mode.IsAppend() || !o.Mode.IsRead() {
+		log.Fatalf("The Object Mode is not appendable")
+	}
+
+	// `o` is the appendable object returned by Stat. It maintains the next call's append position.
+	// User could use o.SetAppendOffset(v) to specify the next append position. If `v` is not equal to the current object length, the service will return an error.
+	n, err := appender.WriteAppend(o, r, size)
```
Comment (Xuanwo) [inline append.go:93]: I think the service should check if the file is `IsAppend`

Comment (JinnyYi) [reply append.go:93]: Yes, I'll delete the mode check.

**Review by Xuanwo — APPROVED** [review]

File: append.go [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+	//
+	// WriteAppend will return two values.
+	// `n` is the next append position. It's valid when `err` is nil.
+	// `err` is the error during this operation.
+	n, err := appender.WriteAppend(o, r, size)
+	if err != nil {
+		log.Fatalf("WriteAppend %v: %v", path, err)
+	}
+
+	log.Printf("append size: %d", n)
```
Comment (Xuanwo) [inline append.go:60]: Please add `CommitAppend` to finish an append process.

**Review by Xuanwo — APPROVED** [review]

**Pull request merged**